src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -570,6 +570,15 @@ internal virtual void ValidateParameters()
                 ErrorRecord error = GetValidationError(WebCmdletStrings.OutFileMissing, "WebCmdletOutFileMissingException", nameof(Resume));
                 ThrowTerminatingError(error);
             }
+
+            _qualifiedOutFile = ShouldSaveToOutFile ? QualifiedOutFile : null;
+
+            // OutFile must not be a directory to use Resume.
+            if (Resume.IsPresent && Directory.Exists(_qualifiedOutFile))
+            {
+                ErrorRecord error = GetValidationError(WebCmdletStrings.ResumeNotFilePath, "WebCmdletResumeNotFilePathException", _qualifiedOutFile);
+                ThrowTerminatingError(error);
+            }
```

Comment:
It doesn't work because, we get the last segment of the url after the redirections.
Example:
`$x = Invoke-WebRequest "https://download.mozilla.org/?product=firefox-stub&os=win"`
`$x.BaseResponse.RequestMessage.RequestUri.Segments[-1] --> Firefox%20Installer.exe`
`?product=firefox-stub&os=win  !=  Firefox%20Installer.exe [DESIRED RESULT]`

We can't obtain the correct name until after the request